common/tracing/context.go

```diff
@@ -0,0 +1,81 @@
+package tracing
```

Comment:
@temaniarpit27 I tried to remove some of complexity from the code introducing a package for typical tracing things. Could you review it and try to extend and use in the PR?

@manav2401 I need your additional review, does the recent changes helped or not.